pyomo/core/base/indexed_component.py

```diff
@@ -803,13 +803,23 @@ def activate(self):
         """Set the active attribute to True"""
         super(ActiveIndexedComponent, self).activate()
         if self.is_indexed():
-            for component_data in itervalues(self):
+            try:
+                component_data_iter = itervalues(self._data)
+            except AttributeError:
+                component_data_iter = iter(self._data)
```

Comment:
Could you add a comment or something here now? It's unclear what attribute it's erroring on.